packages/iTwinUI-react/src/core/Table/actionHandlers/selectHandler.ts

```diff
@@ -88,6 +89,65 @@ export const onSingleSelectHandler = <T extends Record<string, unknown>>(
   return newState;
 };
 
+/**
+ * Handles selection when clicked on a row while shift key is pressed.
+ */
+export const onShiftSelectHandler = <T extends Record<string, unknown>>(
+  state: TableState<T>,
+  action: ActionType,
+  instance?: TableInstance<T>,
+  onSelect?: (
+    selectedData: T[] | undefined,
+    tableState?: TableState<T>,
+  ) => void,
+  isRowDisabled?: (rowData: T) => boolean,
+) => {
+  let startIndex = instance?.flatRows.findIndex(
+    (row) => row.id === state.lastSelectedRow,
+  );
+  let endIndex = instance?.flatRows.findIndex((row) => row.id === action.id);
+
+  if (startIndex == null || startIndex < 0) {
+    startIndex = 0;
+  }
+  if (endIndex == null || endIndex < 0) {
+    endIndex = 0;
+  }
+
+  if (startIndex > endIndex) {
+    const temp = startIndex;
+    startIndex = endIndex;
+    endIndex = temp;
+  }
+
+  const selectedRowIds: Record<string, boolean> = {};
+
+  // 1. Select all rows between start and end
+  instance?.flatRows
+    .slice(startIndex, endIndex + 1)
+    .forEach((r) => (selectedRowIds[r.id] = true));
+
+  // 2. Select all children of the last row (endIndex)
+  if (instance != null) {
```

Comment:
When instance is null? Maybe you can move this check to the beggining of this handler.

Reply:
I moved this check to the top of the file. Is it alright to add a `console.debug()` on [line 106 of this commit](https://github.com/iTwin/iTwinUI-react/pull/774/commits/cb494c594d2d9e274d88d719c9a62974e03983c1#diff-772f25aaf3cd0aa17d5900a4252050c32c48b2efe18b4f6db0d84df08f96cc0bR106)? Or maybe delete that line?

Reply:
You should remove it. It's for internal testing and should not be released in prod version.

Reply:
**[DONE]** Out of curiosity, what is the recommended way in React to log failures in the production version so that we can debug later? Like sometimes when desktop programs or I think even some Android apps crash, they say do you want to send a stack trace to the developer. If we select yes, then I think it sends all the logs about the app crash.

Is there some similar workflow for this in React?